### PR TITLE
[DOCS] Updated contribution docs

### PR DESCRIPTION
**Pull request by TrangPham**
Updates the contribution docs to include some of the information in the old documentation site. Also updates the instructions for installing spark as it can now be installed using just `pip install pyspark`.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: CONTRIBUTING_CODE.md [hunk 1]
```diff
@@ -167,6 +169,8 @@ Run the following command in the root of the `great_expectations` repository to
 pip install -e .
 ```
 
+`-e` will install Great Expectations in “editable” mode. This is not required, but is often very convenient as a developer.
+
```
Comment (Shinnnyshinshin) [inline CONTRIBUTING_CODE.md:173]: 🚀

**Review by Shinnnyshinshin — APPROVED** [review]
This looks great from my end @TrangPham, thank you for the update. It does look like the `cla` step would need to be addressed before merging

File: CONTRIBUTING_CODE.md [hunk 2]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
 
 - Java. See [Java downloads](https://www.java.com/en/download/).
 
-- The PATH or JAVA_HOME environment variables set correctly. See [Setting Java variables in Windows](https://www.ibm.com/docs/en/b2b-integrator/5.2?topic=installation-setting-java-variables-in-windows) or [Setting Java variables in Linux](https://www.ibm.com/docs/en/b2b-integrator/5.2?topic=installation-setting-java-variables-in-linux).
+- The PATH or JAVA_HOME environment variables set correctly. See [Setting Java variables in Windows](https://www.ibm.com/docs/en/b2b-integrator/5.2?topic=installation-setting-java-variables-in-windows) or [Setting Java variables in Linux](https://www.ibm.com/docs/en/b2b-integrator/5.2?topic=installation-setting-java-variables-in-linux). 
+
+On Mac, if you installed Java via the download above, you can set PATH and JAVA_HOME using:
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
On Mac, run the following commands to set the PATH and JAVA_HOME environment variables:
```
image

File: CONTRIBUTING_CODE.md [hunk 3]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
...
+export JAVA_HOME=`/usr/libexec/java_home`
+export PATH=$PATH:$JAVA_HOME/bin 
+```
 
-- Apache Spark installed and configured correctly. See [Spark Overview](https://spark.apache.org/docs/latest/index.html#downloading).
 
 #### Install PySpark
 
-Run the following command to install the PySpark version that matches your version of Apache Spark:
+- Installing PySpark will now install Spark for you. See [Spark Overview](https://spark.apache.org/docs/latest/index.html#downloading).
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
When you install PySpark, Spark is also installed. See [Spark Overview](https://spark.apache.org/docs/latest/index.html#downloading).
```

File: CONTRIBUTING_CODE.md [hunk 4]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
...
 
-```console
-spark-submit --version 2>&1 | grep -o "version [0-9][^ ]*$"
-version 3.2.0
+Run the following command to install the PySpark & Apache Spark:
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
Run the following command to install PySpark and Apache Spark:
```
image

File: CONTRIBUTING_CODE.md [hunk 5]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
...
 
-pip install "pyspark==3.2.0"
+```console
+pip install pyspark
 ```
 
+## If you want to build documentation locally:
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
## Build documentation locally
```

File: CONTRIBUTING_CODE.md [hunk 6]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
...
+
+`pip install -r docs/requirements.txt`
+
+To build documentation, the command is `cd docs; make html`
+
+Documentation will be generated in `docs/build/html/` with the index.html as the index page.
+
+Note: we use autoapi to generate API reference docs, but it’s not compatible with pandas 1.1.0. You’ll need to have pandas 1.0.5 (or a previous version) installed in order to successfully build docs.
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
AutoAPI is used to generate API reference documentation, but it’s not compatible with pandas 1.1.0. To  build the API documentation, install pandas 1.0.5 or earlier.
```
Move this sentence so it appears immediately after the heading.

File: CONTRIBUTING_CODE.md [hunk 7]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
...
+Run the following command to install the PySpark & Apache Spark:
 
-pip install "pyspark==3.2.0"
+```console
+pip install pyspark
 ```
 
+## If you want to build documentation locally:
+
+`pip install -r docs/requirements.txt`
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
Run the following command to review the documentation requirements text file:

```console
pip install -r docs/requirements.txt
```

```

File: CONTRIBUTING_CODE.md [hunk 8]
```diff
@@ -266,21 +270,35 @@ Use the following information to use Spark for code testing.
...
+
+To build documentation, the command is `cd docs; make html`
```
Comment (kwcanuck) [inline CONTRIBUTING_CODE.md]:
```suggestion
Run the follwoing command to build the documentation:

```console
cd docs; make html
```
```

**Review by kwcanuck — APPROVED** [review]

**Comment by kwcanuck** [comment]
@TrangPham if yo can complete the CLA, we can get these changes implemented.

File: CONTRIBUTING_CODE.md (hunk 7 above)
Comment (anthonyburdi) [inline CONTRIBUTING_CODE.md]: I think these instructions are for the legacy docs (in `docs_rtd`). We should probably replace these instructions with instructions on how to build our current docs. I'm hoping to find a bit of time today to simplify that process using an invoke command - let's coordinate.

**Comment by TrangPham** [comment]
@cla-bot check

File: CONTRIBUTING_CODE.md (hunk 7 above)
Comment (TrangPham) [reply CONTRIBUTING_CODE.md]: Ah sounds good. Looks like the docs here already link to the README within docs folder. I will remove this section that is no longer required.

**Comment by TrangPham** [comment]
@anthonyburdi Looks like there's a required docs integration which isn't listed in the checks here: https://github.com/great-expectations/great_expectations/pull/7880/checks

Is there something I am missing?

File: CONTRIBUTING_CODE.md (hunk 7 above)
Comment (anthonyburdi) [reply CONTRIBUTING_CODE.md]: Awesome! For reference, here is the open PR with a new method to build docs locally and updated instructions: https://github.com/great-expectations/great_expectations/pull/7892

**Comment by anthonyburdi** [comment]
> @anthonyburdi Looks like there's a required docs integration which isn't listed in the checks here: https://github.com/great-expectations/great_expectations/pull/7880/checks
> 
> Is there something I am missing?

Great question, I noticed this on contributor PRs as well. I think this has to do with the fact that this PR is from a fork. I've changed the azure triggers to enable "Build pull requests from forks of this repository" for the `docs_integration` pipeline which I think means that pipeline will now run automatically on forks (as intended). I'm not sure if this had been changed recently. Let's keep an eye on it to make sure it is working after enabling on forks.

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Pull request merged**